<launch>

    <!-- Odometry communication (rosserial server) -->
    <node pkg="rosserial_server" type="serial_node" name="rosserial_server_dp" output="screen">
        <param name="port" value="/dev/ttyUSB0" />
        <param name="baud" value="115200" />
        <rosparam>
            require:
                publishers: [ Toposition ]
                subscribers: [ cmd_vel ]
        </rosparam>
    </node>

    <!-- Odometry covariance callback node ( support rqt reconfiguration ) -->
    <node pkg="odometry" type="odometry_node" name="odometry_node" output="screen">
        
        <param name="active" value="true"/>
        <param name="publish" value="true"/>
        <param name="twist_topic" value="/Toposition"/>
        <param name="odom_topic" value="/odom"/>
		<param name="fixed_frame" value="robot1/odom"/>
		<param name="target_frame" value="robot1/base_footprint"/>
        <param name="update_params" value="false"/>
        <param name="covariance_x" value="0.05"/>
        <param name="covariance_y" value="0.05"/>
        <param name="covariance_z" value="0.1"/>
        <param name="covariance_vx" value="0.05"/>
        <param name="covariance_vy" value="0.05"/>
        <param name="covariance_vz" value="0.2"/>
        <param name="covariance_multi_vx" value="0.5"/>
        <param name="covariance_multi_vy" value="0.5"/>
        <param name="covariance_multi_vz" value="0.5"/>
		<param name="using_nav_vel_cb" value="false"/>
		<param name="using_dynamic_reconf" value="true"/>

    </node>

</launch>
